### ISSUE TITLE
Error from referencing a field that has not been defined anywhere appears at the top of the file

### ISSUE THREAD

**Issue by anikaks** [looker-open-source/malloy#292]
Error from referencing a field that has not been defined anywhere appears at the top of the file, instead of appearing where the issue is.

explore: flights is table('malloy-data.faa.flights'){}

query: flights -> {
  aggregate: flight_count
}

<img width="582" alt="Screen Shot 2022-01-28 at 2 48 54 PM" src="https://user-images.githubusercontent.com/7178946/151632863-a8d8ab57-3df8-4be4-88e3-e7cc67411788.png">

**Comment by mtoy-googly-moogly** [comment]
Yeah this is looking up "flight_count" in "flights" and then the error is reported against the definition of flights instead of against the code which contains the reference, this is a mistake and needs to get fixed.

**Comment by mtoy-googly-moogly** [comment]
ok this is actually a different bug. for references which are not definitions, we don't look them up to see if they exist, which means that the error isn't detected until the compiler is gone

i can fix the compiler to catch this
